Philips.CodeAnalysis.MaintainabilityAnalyzers/AvoidTaskResultAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// © 2021 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidTaskResultAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = @"Avoid Task.Result";
+		public const string MessageFormat = @"Methods may not call Result on a Task.";
+		private const string Description = @"To avoid deadlocks, methods may not call Result on a Task.";
+		private const string Category = Categories.Maintainability;
+		private const string HelpUri = @"https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#async-all-the-way";
+
+		private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(Helper.ToDiagnosticId(DiagnosticIds.AvoidTaskResult), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description, helpLinkUri: HelpUri);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+		public string ContainingNamespace = @"System.Threading.Tasks";

```

Comment:
Rather than making these public fields, can you just make them constructor arguments, with a default constructor?

Reply:
Made them private.  Tests do not need them anymore.